torch/nn/parallel/distributed.py

```diff
@@ -395,18 +395,30 @@ def __init__(
 
         self.is_multi_device_module = len({p.device for p in module.parameters()}) > 1
         distinct_device_types = {p.device.type for p in module.parameters()}
-        assert len(distinct_device_types) == 1, (
-            "DistributedDataParallel's input module must be on "
-            "the same type of devices, but input module parameters locate in {}."
-        ).format(distinct_device_types)
+        if len(distinct_device_types) != 1:
+            raise ValueError(
+                "DistributedDataParallel's input module must be on "
+                "the same type of devices, but input module parameters locate in {}.".format(
+                    distinct_device_types
+                )
+            )
         self.device_type = list(distinct_device_types)[0]
 
-        if device_ids is None or self.device_type == "cpu" or self.is_multi_device_module:
-            assert not device_ids and not output_device, (
-                "DistributedDataParallel device_ids and output_device arguments "
-                "only work with single-device GPU modules, but got "
-                "device_ids {}, output_device {}, and module parameters {}."
-            ).format(device_ids, output_device, {p.device for p in module.parameters()})
+        if (
+            device_ids is None
+            or self.device_type == "cpu"
+            or self.is_multi_device_module
+        ):
+            if device_ids or output_device:
+                raise ValueError(
+                    "DistributedDataParallel device_ids and output_device arguments "
+                    "only work with single-device GPU modules, but got "
+                    "device_ids {}, output_device {}, and module parameters {}.".format(
```

Comment:
Unrelated to this PR, but should it say "...and multiple devices/CPU device for module parameters {p.device for p in module.parameters()}" ?

Reply:
Yep, updated! Thanks!